src/com/xilinx/rapidwright/edif/EDIFTools.java

```diff
@@ -1144,32 +1144,44 @@ public static List<String> getMacroLeafCellNames(EDIFCell cell) {
 	
 		return cellInstMap;
 	}
-	
+
 	/**
-	 * Flattens the netlist by creating unique cells within each library except primitives and 
-	 * macros.  It also updates references throughout the logical and physical netlist so all 
-	 * references are self-consistent.  This transformation is useful when performing netlist 
-	 * manipulations such as adding/removing cells, ports or nets within a design.
-	 * @param design The design containing the netlist to flatten.
+	 * Duplicates EDIFCells such that each EDIFCellInst only instantiates an EDIFCell once 
+	 * (except primitives and macros).
+	 * @param design The design containing the netlist to uniqueify.
+	 * @deprecated 
+	 * Please use {@link #uniqueifyNetlist(Design)} instead
+	 * To be removed in 2022.2.0
 	 */
 	public static void flattenNetlist(Design design) {
+	    uniqueifyNetlist(design);
+	}
+	
+	/**
+	 * Duplicates EDIFCells such that each EDIFCellInst only instantiates an EDIFCell once 
+	 * (except primitives and macros).  It also updates references throughout the logical and 
+	 * physical netlist so all references are self-consistent.  This transformation is useful when 
+	 * performing netlist manipulations such as adding/removing cells, ports or nets within a design.
+	 * @param design The design containing the netlist to uniqueify.
+	 */
+	public static void uniqueifyNetlist(Design design) {
 		if(design.getModuleInsts().size() > 0) {
-			System.err.println("ERROR: Cannot flatten netlist, design contains ModuleInstances. "
+			System.err.println("ERROR: Cannot uniqueify netlist, design contains ModuleInstances. "
 					+ "Please call Design.flattenDesign() first.");
 			return;
 		}
 		EDIFNetlist netlist = design.getNetlist();
 		EDIFLibrary macros = Design.getMacroPrimitives(design.getDevice().getSeries());
-		HashMap<EDIFLibrary, HashMap<EDIFCell, ArrayList<EDIFHierCellInst>>> instMap = 
-																	createCellInstanceMap(netlist);
+		Map<EDIFLibrary, Map<EDIFCell, List<EDIFHierCellInst>>> instMap = createCellInstanceMap(netlist);
 		// Don't uniqueify primitive cell instances
 		instMap.remove(netlist.getHDIPrimitivesLibrary());
 		
-		HashMap<EDIFCell, ArrayList<EDIFHierCellInst>> toUniqueify = new HashMap<EDIFCell, ArrayList<EDIFHierCellInst>>();
+		Map<EDIFCell, List<EDIFHierCellInst>> toUniqueify 
+		                                    = new HashMap<EDIFCell, List<EDIFHierCellInst>>();
 		
 		for(EDIFLibrary lib : instMap.keySet()) {
-			HashMap<EDIFCell, ArrayList<EDIFHierCellInst>> cellMap = instMap.get(lib);
-			for(Entry<EDIFCell,ArrayList<EDIFHierCellInst>> e : cellMap.entrySet()) {
+			Map<EDIFCell, List<EDIFHierCellInst>> cellMap = instMap.get(lib);
```

Comment:
Instead of a `keySet()` then `get()` can we do `entrySet()` and `getValue()`?

Reply:
Done.